workflows/alevin-quant/run-alevin-fry.nf

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env nextflow
+nextflow.enable.dsl=2
+
+// run parameters
+params.ref_dir = 's3://nextflow-ccdl-data/reference/homo_sapiens/ensembl-100'
+params.index_dir = 'salmon_index'
+params.index_name = 'txome_k31'
+params.annotation_dir = 'annotation'
+params.t2g = 'Homo_sapiens.ensembl.100.tx2gene.tsv'
+params.mitolist = 'Homo_sapiens.ensembl.100.mitogenes.txt'
+
+params.run_metafile = 's3://ccdl-scpca-data/sample_info/scpca-library-metadata.tsv'
+// run_ids are comma separated list to be parsed into a list of run ids,
+// or "All" to process all samples in the metadata file
+params.run_ids = "SCPCR000001,SCPCR000002"
+
+params.outdir = 's3://nextflow-ccdl-results/scpca/alevin-fry-quant'
+
+// build full paths
+params.index_path = "${params.ref_dir}/${params.index_dir}/${params.index_name}"
+params.t2g_path = "${params.ref_dir}/${params.annotation_dir}/${params.t2g}"
+params.mito_path = "${params.ref_dir}/${params.annotation_dir}/${params.mitolist}"
+
+// generates RAD file using alevin
+process alevin{
+  container 'quay.io/biocontainers/salmon:1.4.0--hf69c8f4_0'
+  label 'cpus_8'
+  tag "${id}-${index}"
+  publishDir "${params.outdir}"
+  input:
+    tuple val(id), path(read1), path(read2)
+    path index
+    path tx2gene
+  output:
+    path run_dir
+  script:
+    run_dir = "${id}-${index}"
+  // run alevin like normal with the --justAlign flag 
+  // creates output directory with RAD file needed for alevin-fry
+    """
+    mkdir -p ${run_dir}
+    salmon alevin \
+      -l ISR \
+      --chromium \
+      -1 ${read1} \
+      -2 ${read2} \
+      -i ${index} \
+      --tgMap ${tx2gene} \
+      -o ${run_dir} \
+      -p ${task.cpus} \
+      --dumpFeatures \
+      --justAlign
+    """
+}
+
+//generate permit list from RAD input 
+process generate_permit{
+  container 'ghcr.io/alexslemonade/scpca-alevin-fry:latest'
+  label 'cpus_8'
+  publishDir "${params.outdir}"
+  input:
+    path run_dir
+  output:
+    path run_dir
+  script: 
+  // expected-ori either signifies no filtering of alignments 
+  // based on orientation, not sure if this is what we want? 
+    """
+    alevin-fry generate-permit-list \
+      -i ${run_dir} \
+      --expected-ori either \
+      -o ${run_dir} \
+      -k
+    """
+}
+
+// given permit list and barcode mapping, collate RAD file 
+// then quantify collated RAD file
+process collate_quant{
```

Comment:
Did you run the workflow [with separating the two steps](https://github.com/AlexsLemonade/alsf-scpca/blob/jashapiro/alevin-fry-sketch/workflows/alevin-quant/run-alevin-fry.nf)? I ask because I had originally put it like this, but I was getting errors where the quant step was having trouble finding the output file from the collate step. So I'm curious if you were able to solve that here? If so then totally good to separate them.

Reply:
Yes, I did run it and it worked fine with me keeping everything in and out as `run_dir`.